networkx/algorithms/coloring/equitable_coloring.py

```diff
@@ -109,7 +109,7 @@ def move_witnesses(src_color, dst_color, N, H, F, C, T_cal, L):
     while X != dst_color:
         Y = T_cal[X]
         # Move _any_ witness from X to Y = T_cal[X]
-        w = [x for x in C[X] if N[(x, Y)] == 0][0]
+        w = next(x for x in C[X] if N[(x, Y)] == 0)
```

Comment:
This line is not protected by try/except block catching `StopIteration` because previous implementation is not catching possible `IndexError` exception. It means that algorithm guarantees (assuming the implementation in main is correct) that it is guaranteed that the list/iterator contains at least one item which can be accessed